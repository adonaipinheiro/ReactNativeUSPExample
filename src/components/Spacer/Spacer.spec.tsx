import React from 'react';

import { act, render } from '@testing-library/react-native';

import { Spacer } from './';

describe('Spacer', () => {
    it('renders a vertical spacer (default)', () => {
        const { getByTestId, toJSON } = render(<Spacer size={8} />);
        const spacer = getByTestId('spacer');

        expect(spacer.props.style).toMatchObject({
            height: 8,
            width: 0,
        });

        expect(toJSON()).toMatchSnapshot();
    });

    it('renders a horizontal spacer when `horizontal` is true', () => {
        const { getByTestId, toJSON } = render(<Spacer horizontal size={12} />);
        const spacer = getByTestId('spacer');

        expect(spacer.props.style).toMatchObject({
            width: 12,
            height: 0,
        });

        expect(toJSON()).toMatchSnapshot();
    });

    it('should re-render if props change', () => {
        const { rerender, getByTestId } = render(<Spacer size={10} horizontal={false} />);

        act(() => {
            rerender(<Spacer size={20} horizontal={true} />);
        });

        const secondRender = getByTestId('spacer');

        expect(secondRender.props.style.width).toBe(20);
        expect(secondRender.props.style.height).toBe(0);
    });

    it('should not re-render if size and horizontal do not change', () => {
        const { rerender, getByTestId } = render(<Spacer size={10} horizontal={false} />);

        act(() => {
            rerender(<Spacer size={10} horizontal={false} />);
        });

        const secondRender = getByTestId('spacer');
        expect(secondRender.props.style.width).toBe(0);
    });
});
