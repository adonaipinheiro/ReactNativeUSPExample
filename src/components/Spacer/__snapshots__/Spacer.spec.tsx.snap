// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders a horizontal spacer when \`horizontal\` is true 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 12,
    }
  }
  testID="spacer"
/>
`;

exports[`Spacer renders a vertical spacer (default) 1`] = `
<View
  style={
    {
      "height": 8,
      "width": 0,
    }
  }
  testID="spacer"
/>
`;
